File: src/main/java/factory/LoggerFactory.java
package factory;

import entities.LoggerConfig;
import impl.AsyncLogger;
import abstracts.Logger;
import enums.LoggerType;
import impl.SyncLogger;

public class LoggerFactory {

    public Logger getLogger(LoggerConfig config){
        return switch (config.loggerType){
            case LoggerType.SYNC -> new SyncLogger(config);
            case ASYNC -> new AsyncLogger(config);
        };
    }
}

File: src/main/kotlin/factory/LoggerFactory.kt
package factory

import entities.Logger
import entities.LoggerConfig
import entities.LoggerType
import impl.AsyncLogger
import impl.SyncLogger

class LoggerFactory {
    fun getLogger(config: LoggerConfig): Logger {
        return when (config.loggerType) {
            LoggerType.SYNC -> SyncLogger(config)
            LoggerType.ASYNC -> AsyncLogger(config)
        }
    }
}
